community/cypher/cypher/src/main/java/org/neo4j/cypher/internal/codegen/CompiledMaterializeValueMapper.java

```diff
@@ -106,9 +105,9 @@ public AnyValue mapRelationship( VirtualRelationshipValue value )
         @Override
         public AnyValue mapMap( MapValue value )
         {
-            Map<String,AnyValue> map = new HashMap<>();
-            value.foreach( ( k, v ) -> map.put( k, v.map( this ) ) );
-            return VirtualValues.map( map );
+            MapValueBuilder builder = new MapValueBuilder();
```

Comment:
we could initialize this new ```MapValueBuilder``` with ```value.size```, right?

Reply:
`value.size()` may be a O(n) operation so I think I prefer not to do it in this case.